src/frontend/js/modules/table-of-content.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Generate dynamic table of content
+ */
+export default class TableOfContent {
+  /**
+   * Initialize table of content
+   *
+   * @param {string} tagSelector - selector for tags to observe
+   * @param {string} tocWrapperSelector - selector for table of content wrapper
+   */
+  constructor({ tagSelector, tocWrapperSelector }) {
+    this.tagSelector = tagSelector || 'h2,h3,h4';
+    this.tocWrapperSelector = tocWrapperSelector;
+
+    this.init();
+  }
+
+  /**
+   * Initialize table of content
+   */
+  init() {
+    this.findTagsOnThePage();
```

Comment:
The name is not clear. Better name method in business terms like `calculateHeadingCoords` or something like that